### ISSUE TITLE
No quitters option on table creation

### ISSUE THREAD

**Issue by menocar** [magefree/mage#1499]
We can add such an optipn to restrict participants. Those who are fine with quits, or new users, can still play without setting that option.

Here is my initial idea whitch is to offer two levels:

* Default: you need to have finished at least one game (a match or a tourney, depending on what the table is for), and your quits needs to be no more than 20% of the matches played.
* Strict: same as the default, but 10% instead of 20%.

Alternativey we can check the non-quitting streak, though it's easily affected by accidental quits.

**Comment by LevelX2** [comment]
Not setting is like setting it to 100%, 
So maybe it's the most flexible to only use one % value that you can set at the level you like?

**Comment by LevelX2** [comment]
Conceding is now handled as not quitting?
I was always a bit upset if I drafted and constructed a deck and as soon as the players were paired, my opponent conceded after 5 seconds, next round having a bye doesn't make this better. :-)

So we should think about how we handle such short games. 
Maybe if duration of a match is shorter than x minutes the winner has to agree, that the conceding player gets no quit rate penalty can be a solution?

**Comment by LevelX2** [comment]
Also we need something like a calculated "Quit" rate, that should be added to the user list as column as well. So sorting will work and you can get the value without calculating with the history values.

**Comment by menocar** [comment]
Yeah, I thought it's great if we can present some predefined configuration so that users try not to exceed a certain percentage, but maybe we should provide one flexible option and let users decide how much strict they want to be.

Regarding quick concedes: we should definitely start recording match time. I think we can simply treat them as quits, because practically the end result is the same. I guess 1 minute is a good threshold.

Showing quit rate separately sounds good. BTW I just figured instead of the current single History column it's more concise to have two columns named Matches and Tourneys. For example,

History
Matches:26 (Q:6) Tourneys:10 (D:2 R:1)

becomes

Matches | Tourneys
26 (Q:6) | 10 (D:2 R:1)

I'd like to make the change when I have time.

**Comment by LevelX2** [comment]
And showing the quit-rate with a % value would be helpful, if we use such a value to allow the filtering.
Maybe leading zeroes to allow sorting by the number of matches/tourneys?
Quit% | Matches | Tourneys
03% | 00100 (Q:1) | 00005 
23% | 00026 (Q:6) | 00010 (D:2 R:1)

**Comment by menocar** [comment]
I hope there's a way to specify custom sort functions without needing to invent a good formatting :)  We should fix the sorting for network latency column too.

**Comment by fireshoes** [comment]
I'm sure it is on the long list of to-do's, but I did have a match yesterday where I lost connection. The pop-up messages came up "Do you want to reconnect?" I clicked yes. "Ping failed" message. Clicked ok. I get brought back to my match and the "Do you want to reconnect?" message comes back up. I click "Yes" and it disconnects me and quits the match (oh no, my perfect quit rating! lol). It's not something that happens all the time, but I see it occur pretty regularly. You will notice an opponent lost connection, then they rejoin, then quit the match also, so you can tell it happens to others. Anyway, something getting goofed up in the reconnect dialog that would be great to have fixed by people smarter than me. :D

**Comment by LevelX2** [comment]
What about dropping thw counted quits after a certain time span? (mayby all older than x month)

**Comment by menocar** [comment]
Yeah, unintentional disconnects makes things complicated. I think we have to bear that until someone can fix the issue. We all know disconnects occur with a certain probability, so you shouldn't feel too bad about small quit counts. All unti-quitter functionality should take that into account. In case a user exceeds the limit by purely unintentional disconnects... then maybe the user needs to consider improving the network environment.

**Comment by menocar** [comment]
Sent https://github.com/magefree/mage/pull/1508

**Comment by menocar** [comment]
The current code simply checks quit ratio so it accepts new users. Probably we want to also check if they play enough games? For example, if quit ratio is set to X%, those who played less than (100/X)-1 games can't join.

**Comment by LevelX2** [comment]
Maybe it would be best to have an additional option to switch the check for not enough matches to join on/off.

**Comment by spjspj** [comment]
Another definite problem with playing people is if they 'play slow' deliberately.  Aka Person B gets salty, wait 5 minutes, does an action, wait 5 minutes, come back and does another action.  Another problem is that people can follow/stalk(?) the person they got salty with to each new table they create.  A per-table ban list may be useful (aka, Person A doesn't want a password on the table, but does want to not allow Person B to join it).

**Comment by LevelX2** [comment]
Maybe for tournaments it could be a good solution to some of taht problems to give the tournament creator some extra rights like an organizator of an tournament.
A) The creator should be able to only create the table and should not be forced to play the tournament.
B) Should be able to ban players to join or remove players from the table.
C) Should be able to to cancel games/matches and set the outcome.
D) Should be able to change the outcome of a match.
E) Be able to watch a game even if watching is not allowed.

And sure, each user should be able to create a personal group of players that are not allowed to join tables he creates.

**Comment by fireshoes** [comment]
That sounds great.

**Comment by Zerrisx** [comment]
Can we rename and/or remake this issue?  The original issue is no longer what's being discussed here.

**Comment by sumsarz** [comment]
Not being able to set a limit on amount of games someone has (which I'd deem is related to this topic) is abused by people on the beta server to just make another account to join games after they've been ignored.
